plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1757,13 +1755,21 @@ public function trash_order( $order ) {
 
 		$wpdb->update(
 			self::get_orders_table_name(),
-			array( 'status' => 'trash' ),
+			array(
+				'status'           => 'trash',
+				'date_updated_gmt' => current_time( 'Y-m-d H:i:s', true ),
+			),
 			array( 'id' => $order->get_id() ),
-			array( '%s' ),
+			array( '%s', '%s' ),
 			array( '%d' )
 		);
 
 		$order->set_status( 'trash' );
+
+		$data_synchronizer = wc_get_container()->get( DataSynchronizer::class );
+		if ( $data_synchronizer->data_sync_is_enabled() ) {
+			wp_trash_post( $order->get_id() );
```

Comment:
> Untrash the order and verify that the order has the proper status and that there aren't _wp_trash_meta_status entries in the wp_wc_orders_meta table for the order.

☝🏽 This part is not working for me (COT authoritative, sync enabled).

It looks to me like we store the pre-trash status and trash time as order meta data, just [a few lines above](https://github.com/woocommerce/woocommerce/pull/35125/files#diff-8c34c1a6f743002d1f2db13c299d2086d0209889f72abc86bef3bd734abb2585R1741-R1754) this point. Then execution reaches here and we trash the corresponding post record ... the problem is that those bits of meta data never makes it across to the post meta table. The result is the order is not restored to its original status when it is untrashed.

Steps to replicate:

1. Check out this branch, set COT to authoritative, enable sync.
2. Select an order that is **not** *pending payment,* note its status and trash it.
3. Visit the trashed orders view.
4. Untrash the order, note it will have been incorrectly restored to *pending payment* status.

Reply:
I can't reproduce this. I have tried with "on hold", "processing" and "complete" and in all cases:

1. Trashing the order generates the proper meta items in both meta tables:

![image](https://user-images.githubusercontent.com/937723/196370650-4fa12172-1132-4c9c-a70b-056156bf109f.png)

2. Untrashing the order restores it to the proper state.

What I have noticed is that after untrashing the entry in `wp_postmeta` is still there. Maybe because it doesn't have the `wc-` prefix in the status name?

Reply:
Interesting ... I wonder why the difference. Let me dig just a little bit further and/or verify using a fresh/'blank' environment. Perhaps something particular to me environment is causing an issue.

Reply:
I still see this issue. I'll send you some details to help you see it.

Reply:
Fixed. Explanation of what was going on in [the commit message](https://github.com/woocommerce/woocommerce/pull/35125/commits/f538c7c272fa5c02010aea62a281462cf5b8dccb).

Reply:
Yep, works :+1: